Source/WebCore/platform/graphics/ContentsFormat.h

```diff
@@ -42,6 +42,8 @@ enum class ContentsFormat : uint8_t {
 #endif
 #if ENABLE(PIXEL_FORMAT_RGBA16F)
     RGBA16F = 1 << 2,
+    RGBA16FConstrained = 1 << 3,
+    RGBA16FSDR = 1 << 4,
```

Comment:
What is the meaning of `RGBA16F` after `RGBA16FConstrained` and `RGBA16FSDR`? Does it actually mean `RGBA16FHigh` or `RGBA16FNoLimit`? If yes so then these new enum are like combining `ContentsFormat` and `DynamicRangeLimit` (without the mix function) in one enum value.

I think `ContentsFormat` should describe the backing store of the layer only. It should describe whether is 8, 10 or 16 bits per channel. It should not include other information about tone-mapping the pixels to a certain `DynamicRangeLimit`.

Can't we add another member of type `DynamicRangeLimit` or `PlatformDynamicRangeLimit` to control suppressing the HDR-ness of the `RGBA16F` layer?

Reply:
Yes that is certainly a good option. I wasn’t sure if we needed to keep the sizeof(LayerProperties) small or if the additional enum byte is fine.

I’ll refactor the patch to add DynamicRangeLimit